.travis/before_install.sh

```diff
@@ -120,12 +120,23 @@ case "${BUILD_TARGET}" in
 
         cd $HOME
         git clone https://github.com/openweave/happy.git
-
+        echo "=========>copy happy/bin files======>"
```

Comment:
it is because make install also does install happy package in system's python env and create some folder as root permission under happy folder.
when it comes to install happy in the virtual env, it will have permission issue for virtual env to install happy.

Reply:
Do we strictly need virtualenv? This all seems to run fine without virtualenv on my Ubuntu machine and, from what I can see, Travis CI instances are mostly Debian installations.

If this is necessary, it seems like this part of our before_install.sh script (or somewhere else) in our system needs some heavier documentation as to these nuanced exceptions from practice documented elsewhere. Otherwise, this all ends up seeming like black magic or tribal knowledge to the uninitiated.

Reply:
I checked with @yunhanw-google yesterday, it is not strictly need virtualenv for travis-CI existing tests. but when it come to service tests, we need virtualenv
I agree it is black magic around here:), it took me a while to figure out the root cause of the "happy commands not found" error yesterday from Travis CI test run.
I am OK if we want to switch everything without using virtual env.
@yunhanw-google, could you explain why do we need virtualenv for service tests?